File: python/01-commands/doc/Banking_Kata/account_test.py
import unittest
from account import Account

class AccountTest(unittest.TestCase):
    def test_deposit_increases_balance(self):
        # Arrange
        deposit_amount = 500
        account = Account()

        # Act
        account.deposit(deposit_amount)

        # Assert
        self.assertEqual(deposit_amount, account.balance)

    def test_withdraw_decreases_balance_and_updates_statement(self):
        # Arrange
        account = Account()
        withdraw_amount = 500

        # Act
        account.withdraw(withdraw_amount)

        # Assert
        self.assertEqual(-withdraw_amount, account.balance)

    def test_withdraw_after_deposit_updates_balance_and_statement(self):
        # Arrange
        account = Account()
        deposit_amount = 500

        # Act
        # The rest of the code is missing

if __name__ == '__main__':
    unittest.main()

File: dotnet/01-commands/doc/Banking_Kata/AccountTest.cs
namespace Banking_Kata
{
    public class AccountTest
    {
        [Fact]
        public void WhenADepositInMade_BankAccountInceasesByThisAmount()
        {
            //Arrange 
            int depositAmount = 500;
            Account account = new Account();

            //Act
            account.Deposit(depositAmount);

            //Assert
            Assert.Equal(depositAmount, account.balance);
        }

        [Fact]
        public void WhenAWithdrawIsDone_BankAccountDecreasesByThisAmountAndStatementUpdated()
        {
            //Arrange 
            Account account = new Account();
            int withdrawAmount = 500;

            //Act
            account.Withdraw(withdrawAmount);

            //Assert
            Assert.Equal(-withdrawAmount, account.balance);
        }

        [Fact]
        public void WhenAWithdrawAfterDepositIsDone_BankAccountShouldBeUpdated()
        {
            //Arrange 
            Account account = new Account();
            int depositAmount = 500;
            int withdrawAmount = 500;

            //Act
            account.Deposit(depositAmount);
            account.Withdraw(withdrawAmount);

            //Assert
            Assert.Equal(depositAmount -withdrawAmount, account.balance);
        }

        [Fact]
        public void WhenADepositInMade_BankIsUpdatedByThisAmount()
        {
            //Arrange 
            int depositAmount = 500;
            Account account = new Account();
            List<BankingTransactionDTO> expectedStatement = new List<BankingTransactionDTO>();
            expectedStatement.Add(new BankingTransactionDTO(DateTime.Today, depositAmount, depositAmount));

            //Act
            account.Deposit(depositAmount);

            //Assert
            Assert.Equal(expectedStatement, account.statement);
        }

        [Fact]
        public void WhenAWithdrawIsMade_BankIsUpdatedByThisAmount()
        {
            //Arrange 
            int withdrawAmount = 500;
            Account account = new Account();
            List<BankingTransactionDTO> expectedStatement = new List<BankingTransactionDTO>();
            expectedStatement.Add(new BankingTransactionDTO(DateTime.Today, -withdrawAmount, -withdrawAmount));

            //Act
            account.Withdraw(withdrawAmount);

            //Assert
            Assert.Equal(expectedStatement, account.statement);
        }
    }
}